exporters/otlp-http/trace/src/test/java/io/opentelemetry/exporter/otlp/http/trace/OtlpHttpSpanExporterTest.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.exporter.otlp.http.trace;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import io.github.netmikey.logunit.api.LogCapturer;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.exporter.otlp.internal.SpanAdapter;
+import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.common.InstrumentationLibraryInfo;
+import io.opentelemetry.sdk.testing.trace.TestSpanData;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.data.StatusData;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import okhttp3.mockwebserver.MockResponse;
+import org.junit.Rule;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.slf4j.event.Level;
+import org.slf4j.event.LoggingEvent;
+
+class OtlpHttpSpanExporterTest {
+
+  @Rule private final OtlpHttpRule otlpHttp = new OtlpHttpRule();
```

Comment:
There seems to be a mix of junit5 and junit4 - does it work? Anyways we'd prefer to stick with junit5 - can you use the armeria mock server we have in a couple other tests?

https://github.com/open-telemetry/opentelemetry-java/blob/c3614b6b2345361a3484f92961231ca19b558242/sdk-extensions/aws/src/test/java/io/opentelemetry/sdk/extension/aws/internal/JdkHttpClientTest.java

Not sure but don't think you should need a custom dispatcher either, the mock server takes care of allowing assertions etc

Reply:
It did work, but yeah that was definitely an oversight. I've switched to the armeria mock server - haven't used it before but I like the interface! Definitely simplifies stuff nicely.